config/reader.go

```diff
@@ -29,6 +31,14 @@ var (
 	GitRevision = ""
 )
 
+var (
+	placeholderRegexp = regexp.MustCompile(`\$\{(packages[\.(ml_model|modules)]*\.[A-Za-z0-9_\-]+)\}`)
```

Comment:
changed this a lil bit to only allow at most 1 ml_model/module but no this will replace packages.name too. It checks if 
0 or 1 [\.(ml_model|modules)] exists as the 2nd directory so it allows for a custom name here. If there is a custom name instead of ml_models/modules then if there is another subdirectory this will fail. 

E.g. ${packages,name} => packages/name
E.g. ${packages.ml_models.name} => FAIL

I also have tests to validate all 3 use cases which are working.